### PR TITLE
Fix hlint hints.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: src/Test/StateMachine/Utils.hs [hunk 1]
```diff
@@ -51,10 +50,11 @@ forAllShrinkShow gen shrinker shower pf =
 forAllShow
   :: Testable prop
   => Gen a -> (a -> String) -> (a -> prop) -> Property
-forAllShow gen shower pf = forAllShrinkShow gen (\_ -> []) shower pf
+forAllShow gen = forAllShrinkShow gen (const [])
```
Comment (Danten) [inline src/Test/StateMachine/Utils.hs]: Oh so when `hlint` says it you change it...

File: src/Test/StateMachine/Internal/Sequential.hs [hunk 2]
```diff
@@ -150,7 +150,7 @@ liftSem
 liftSem sem cmd iref = do
 
   env <- get
-  let cmd' = ifmap @_ @_ @_ @_ @refs (\s i -> env IxM.! (s, i)) cmd
+  let cmd' = ifmap @_ @_ @_ @_ @refs (curry ((IxM.!) env)) cmd
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: I'm not sure I agree with this one.

File: src/Test/StateMachine/Internal/Parallel.hs [hunk 3]
```diff
@@ -117,8 +118,7 @@ liftShrinkFork shrinker f@(Fork l0 p0 r0) =
       r'' = removeCommands p r
 
       removeManyCommands :: [IntRefed cmd] -> [IntRefed cmd] -> [IntRefed cmd]
-      removeManyCommands []       ds = ds
-      removeManyCommands (c : cs) ds = removeManyCommands cs (removeCommands c ds)
+      removeManyCommands cs ds = foldl (flip removeCommands) ds cs
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Parallel.hs]: I think it is easier to see what the recursive functions does.

File: src/Test/StateMachine/Utils.hs (hunk 1 above)
Comment (stevana) [inline src/Test/StateMachine/Utils.hs]: ;-)

File: src/Test/StateMachine/Internal/Sequential.hs (hunk 2 above)
Comment (stevana) [inline src/Test/StateMachine/Internal/Sequential.hs]: Me neither, but then we need to figure out how to disable these kind of hints...

File: src/Test/StateMachine/Internal/Parallel.hs (hunk 3 above)
Comment (stevana) [inline src/Test/StateMachine/Internal/Parallel.hs]: Yeah...

**Comment by stevana** [comment]
I've reverted the two hints that produced less readable code.

**Pull request merged**